src/render/OpenGL.cpp

```diff
@@ -1686,28 +1686,34 @@ void CHyprOpenGLImpl::createBGTextureForMonitor(CMonitor* pMonitor) {
     // or configure the paths at build time
 
     // get the adequate tex
-    std::string texPath = "/usr/share/hyprland/wall_" + std::string(USEANIME ? (distribution2(engine) == 0 ? "anime_" : "anime2_") : "");
+    std::string texName = "wall_" + std::string(USEANIME ? (distribution2(engine) == 0 ? "anime_" : "anime2_") : "");
     // check if wallpapers exist
 
     Vector2D textureSize;
     if (pMonitor->vecTransformedSize.x > 3850) {
         textureSize = Vector2D(7680, 4320);
-        texPath += "8K.png";
+        texName += "8K.png";
     } else if (pMonitor->vecTransformedSize.x > 1930) {
         textureSize = Vector2D(3840, 2160);
-        texPath += "4K.png";
+        texName += "4K.png";
     } else {
         textureSize = Vector2D(1920, 1080);
-        texPath += "2K.png";
+        texName += "2K.png";
     }
 
-    if (!std::filesystem::exists(texPath)) {
-        // try local
-        texPath = texPath.substr(0, 5) + "local/" + texPath.substr(5);
-
-        if (!std::filesystem::exists(texPath))
-            return; // the texture will be empty, oh well. We'll clear with a solid color anyways.
-    }
+    // goto is banned? ok have a do while.
+    std::string texPath = "/usr/share/hyprland/" + texName;
+    do {
```

Comment:
wtf?

if ()
 path =
else if ()
 path = 
 
 ??

Reply:
That won't actually work. Its has to be a bunch of nested ifs, a do while (false), or a goto.

Reply:
oh really?

```cpp
auto findPath = [&] (std::string& path) -> void {
   if (std::filesystem::exists(path)
        return;
   path = "/usr/local/share/hyprland/" + texName;
   if (std::filesystem::exists(path))
       return;
   path = "assets/" + texName;
   if (std::filesystem::exists(path))
       return;
    path = "";
};

findPath(texPath);
```

Reply:
That also works I guess

Reply:
Fixed